indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java

```diff
@@ -219,6 +221,7 @@
   private final List<ListenableFuture<SegmentsAndMetadata>> handOffWaitList = new ArrayList<>();
 
   private final LockGranularity lockGranularityToUse;
+  private final ExecutorService sequencePersistExecutor;
```

Comment:
In classes that don't have `close()`/`stop()` methods yet, like `SeekableStreamIndexTaskRunner`, they should be introduced if you want to manage an executor.

If the class is pretty ephemeral (e. g. many instances of this class routinely start and stop), it should be strongly preferred to create an executor one level up the stack and reuse it for all instances of the class (or all instances of the class within a certain domain e. g. all `SeekableStreamIndexTaskRunner`s associated with a single datasource, or something like that).

Note again, I don't actually tell you that this class `SeekableStreamIndexTaskRunner` falls into this category - please figure it out yourself, or consult the maintainers of this subsystem.